src/jit/CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ set( JIT_SOURCES
   bitset.cpp
   block.cpp
   codegencommon.cpp
+  codegenlinear.cpp
```

Comment:
I think that these are excellent points:

> - I think we want more, smaller source files rather than larger ones.
> - I think we are better off not combining legacy and non-legacy in the same file, at least going forward. If anything, we should consider moving the ryujit-only methods from codegencommon.cpp to codegenlinear.cpp.

One of the things that I find difficult about navigating the source base is dealing with the enormous files and the `#ifdef` hell we often find ourselves in. Separating things into smaller, target-specific or target-independent files is a nice way to alleviate this pain.